db/migrate/20210416143448_add_full_text_search_to_supporters.rb

```diff
@@ -0,0 +1,30 @@
+class AddFullTextSearchToSupporters < ActiveRecord::Migration
+  def up
+    add_column :supporters, :fts, :tsvector
+
+    execute(<<-'eosql'.strip)
+      UPDATE supporters SET fts = (to_tsvector('english', coalesce(name, '') || ' ' || coalesce(email, '')));
+
+      CREATE FUNCTION update_fts_on_supporters() RETURNS TRIGGER AS $$
+        BEGIN
+          new.fts = to_tsvector('english', coalesce(new.name, '') || ' ' || coalesce(new.email, ''));
+          RETURN new;
+        END
+      $$ LANGUAGE plpgsql;
+
+      CREATE TRIGGER update_supporters_fts BEFORE INSERT OR UPDATE
+        ON supporters FOR EACH ROW EXECUTE PROCEDURE update_fts_on_supporters();
+
+      CREATE INDEX supporters_fts_idx ON supporters USING gin(fts);
```

Comment:
On staging the migration took 23.0457s for donations_fts_idx and 103.0587s for supporters_fts_idx. If we are going to migrate to Rails 4, I think we can wait to merge this when the new version is ready because we can use `disable_ddl_transaction!` :)